File: 0009-Palindrome-Number/solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x % 10 == 0 and x != 0: return False
        if x < 0: return False

        reverse = 0

        while x > reverse:
            reverse = reverse * 10 + x % 10
            x //= 10

        return reverse == x or reverse // 10 == x

File: 0009-Palindrome-Number/solution.c
bool isPalindrome(int x) {

    if (x % 10 == 0 && x != 0) return false;
    if (x < 0) return false;

    int reverse = 0;

    while (x > reverse) {
        reverse = reverse * 10 + x % 10;
        x /= 10;
    }
    
    if (reverse > x) {
        reverse /= 10;
    }

    return (reverse == x);
}
